### PR TITLE
build: Update to Go 1.20, Alpine 3.17 and linter v1.51.2

### PR DESCRIPTION
**Pull request by lenny-intel**
**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-virtual-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?) **N/A**
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) **N/A**
      <link to docs PR>

## Testing Instructions
verify make docker works
Run non-secure EdgeX stack w/o Device Virtual
Build and run service from command line
Verify logs have no errors bootstrapping.
Verify that auotevents are triggering and Events are pushed to the MessageBus

### CODE REVIEW

File: internal/driver/helper.go [hunk 1]
```diff
@@ -86,6 +86,7 @@ func randomInt(min, max int64) int64 {
 }
 
 func randomUint(min, max uint64) uint64 {
+	//nolint // SA1019: rand.Seed has been deprecated
```
Comment (lenny-intel) [inline internal/driver/helper.go:89]: Wasn't able to find a good alternative and the fact that this is a sample device service chose to ignore the lint error.

Comment (bnevis-i) [reply internal/driver/helper.go:89]: Good enough.

**Review by bnevis-i — APPROVED** [review]

**Review by cloudxxx8 — APPROVED** [review]
LGTM

File: internal/driver/helper.go (hunk 1 above)
Comment (cloudxxx8) [reply internal/driver/helper.go:89]: I open an issue for that, and we can deal with it in the future
https://github.com/edgexfoundry/device-virtual-go/issues/358

**Pull request merged**